lib/charms/layer/kubernetes_common.py

```diff
@@ -572,6 +572,9 @@ def generate_openstack_cloud_config():
         ]
     )
 
+    if openstack.lb_enabled is False:
+        lines.append("enabled = false")
+
```

Comment:
totally nit-picky -- i noticed in the [interface code](https://github.com/juju-solutions/interface-openstack-integration/pull/13/files#diff-8929a787830aa5348f2cd5bb817c7731c3ced0c77fd11d914b036b194126c87bR123) that the default is `None`.  Now, the config driving this comes from the [integrator](https://github.com/juju-solutions/charm-openstack-integrator/pull/67/files#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034R56-R61) which is a `boolean` -- but there's nothing in the relation schema forcing this to be `False` or `True`

would you rather:

```suggestion
    if not openstack.lb_enabled:
        lines.append("enabled = false")
```

or change the interface code to be:

```python
    @property
    def lb_enabled(self) -> bool:
        """
        Whether or not LoadBalancer service integration should be enabled in
        openstack-cloud-controller-manager.
        """
        return bool(self._received['lb_enabled'])
```

Reply:
My concern is that `lb_enabled` may not exist in the relation data if kubernetes-control-plane is related to an old revision of openstack-integrator. If the remote hasn't explicitly told us to disable LB support, then I think we should enable it by default, to preserve backward compatibility and ensure LB support doesn't get temporarily disabled during upgrades.

If `lb_enabled` isn't set in the relation data, then `self._received['lb_enabled']` will return `None` (supporting code/docs: [1](https://github.com/juju-solutions/interface-openstack-integration/blob/aa365314041ccbd0018e7c73e0de39eed9be045f/requires.py#L57-L63), [2](https://github.com/juju-solutions/charms.reactive/blob/df24f1a80c66a5ced87e922d35a0968b7137c56c/charms/reactive/endpoints.py#L161-L164), [3](https://github.com/juju-solutions/charms.reactive/blob/df24f1a80c66a5ced87e922d35a0968b7137c56c/charms/reactive/endpoints.py#L393-L406), [4](https://github.com/juju-solutions/charms.reactive/blob/df24f1a80c66a5ced87e922d35a0968b7137c56c/charms/reactive/endpoints.py#L809-L812), [5](https://github.com/juju-solutions/charms.reactive/blob/df24f1a80c66a5ced87e922d35a0968b7137c56c/charms/reactive/endpoints.py#L875-L880), [6](https://github.com/juju-solutions/charms.reactive/blob/df24f1a80c66a5ced87e922d35a0968b7137c56c/charms/reactive/endpoints.py#L860-L861)). So `bool(self._received['lb_enabled'])` does not work because `None` is falsey, but we want the result in that case to be `True`.

What about something like this? Would putting this in the interface, combined with the suggested change above, resolve your concern?

```python
    @property
    def lb_enabled(self) -> bool:
        """
        Whether or not LoadBalancer service integration should be enabled in
        openstack-cloud-controller-manager.
        """
        # be careful to ensure a None value is returned as True, for backwards compatibility
        return self._received['lb_enabled'] is not False
```

Reply:
Yeah, i think i really like this in the interface since `self._received['lb_enabled']` is `None` by default which should indicate that we're enabled (as it's the backwards compatible setting)

```python
    @property
    def lb_enabled(self) -> bool:
        """
        Whether or not LoadBalancer service integration should be enabled in
        openstack-cloud-controller-manager.
        """
        # be careful to ensure a None value is returned as True, for backward compatibility
        return self._received['lb_enabled'] is not False
```

If so -- leave this line of code like this:

```suggestion
    if openstack.lb_enabled is False:
        lines.append("enabled = false")
```

Reply:
OK, sounds good. Updated interface in https://github.com/juju-solutions/interface-openstack-integration/pull/13, will leave this as is.